src/obu.rs

```diff
@@ -2132,21 +2124,22 @@ unsafe fn parse_obus(
 
     // obu header
     gb.get_bit(); // obu_forbidden_bit
-    let r#type = gb.get_bits(4) as Rav1dObuType;
-    let has_extension = gb.get_bit() as c_int;
-    let has_length_field = gb.get_bit() as c_int;
+    let raw_type = gb.get_bits(4);
+    let r#type = Rav1dObuType::from_repr(raw_type as usize);
+    let has_extension = gb.get_bit();
+    let has_length_field = gb.get_bit();
```

Comment:
a general question: how important is the performance of `GetBits`? its `refill` method only reads data one byte at a time, which in microbenchmarks is quite inefficient. But maybe in practice this is just not a performance-sensitive part of the code?

This is the logic we've taken from `zlib-ng`: https://github.com/memorysafety/zlib-rs/blob/main/zlib-rs/src/inflate/bitreader.rs#L103

Reply:
I don't believe it's very performance critical.  I remember @randomPoison did some benchmarks and I don't think it mattered much, since it's just used to parse the headers.  It would be not too hard to swap out the implementation, though, like the approach you took, but we did already make it safe.  I also looked into `bitvec` a bit but the generated code didn't look as optimized.  I was also hoping to make APIs like `.get_bits` const generic and return `uN` like types, as that also avoids bounds checking for a bunch of arrays.  It sucks that Rust doesn't have `uN` native types, though.  I explored that a bit but it wasn't a priority since we already made it safe, but once we finish safety, I'd like to revisit it to squeeze some more performance out.